### PR TITLE
Add support for composition data page 1.

### PR DESCRIPTION
**Pull request by wiktorzbroja-silvair**

### CODE REVIEW

File: bluetooth_mesh/models/models.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ async def get_composition_data(
         send_interval: float = 2.0,
         progress_callback: Optional[ProgressCallback] = None,
         timeout: float = None,
-    ) -> CompositionData:
+    ) -> CompositionDataPage0Element:
```
Comment (michallowasrzechonek-silvair) [inline bluetooth_mesh/models/models.py:262]: Page0Element or Page0?

Comment (wiktorzbroja-silvair) [reply bluetooth_mesh/models/models.py:262]: `CompositionDataPage0`, thanks!

File: bluetooth_mesh/messages/config.py [hunk 2]
```diff
@@ -543,25 +543,66 @@ def AddressTypeValidator(subcons, *allowed_types):
...
+    "sig_models" / ModelRelationItem[this["number_s"]],
+    "vendor_models" / ModelRelationItem[this["number_v"]],
+)
+
+CompositionDataPage1 = Struct(
+    'element' / GreedyRange(CompositionDataPage1Element)
 )
 
 Retransmit = BitStruct(
-    # sssssccc
```
Comment (kamiljanowski-silvair) [inline bluetooth_mesh/messages/config.py:564]: This is not related to this PR, i prefer to keep it at is for better understanding of bitfield structure in this case

File: bluetooth_mesh/messages/config.py [hunk 3]
```diff
@@ -735,8 +776,8 @@ class CompositionDataPage(enum.IntEnum):
 ConfigCompositionData = Switch(
     this.page,
     {
-        CompositionDataPage.ZERO: CompositionData,
-        CompositionDataPage.FIRST: GreedyBytes,
+        CompositionDataPage.ZERO: CompositionDataPage0,
+        CompositionDataPage.FIRST: CompositionDataPage1,
         CompositionDataPage.TWO_HUNDRED_AND_FIFTY_FIFTH: GreedyBytes,
```
Comment (kamiljanowski-silvair) [inline bluetooth_mesh/messages/config.py:781]: add `default: GreedyBytes` for other, not supported pages

File: bluetooth_mesh/messages/test/test_config.py [hunk 4]
```diff
@@ -472,16 +549,51 @@
...
+                                "corresponding_id": 0x0A,
+                                "extended_models_items": []
+                            }
+                        ],
+                        "vendor_models": []
+                    }
+                ]
+                }
         },
-        id="ConfigCompositionDataStatus - not page 0"
```
Comment (kamiljanowski-silvair) [inline bluetooth_mesh/messages/test/test_config.py:484]: I prefer to keep a UT for other pages that shall be parsed as GreedyBytes

**Review by kamiljanowski-silvair — APPROVED** [review]
